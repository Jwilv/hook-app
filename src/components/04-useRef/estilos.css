body{
    padding: 70px;
}
button{
    margin: 15px;
}