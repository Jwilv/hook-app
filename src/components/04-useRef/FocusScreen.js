import React, { useRef } from 'react'
import './estilos.css'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = ()=>{
    //document.querySelector('input').select();
    inputRef.current.select();
    }


    return (
        <>
        <h1>Focus Screen</h1>
        <hr/>
        <input
        ref={inputRef}
        className='form-control'
        placeholder='su nombre'
        /> 
        <button
        onClick={handleClick} 
        className='btn btn-primary'
        >Focus</button>
        </>
    )
}
