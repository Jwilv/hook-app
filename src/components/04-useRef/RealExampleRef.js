import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-example/MultipleCustomHooks'
import './estilos.css'

export const RealExampleRef = () => {

  const [show, setShow] = useState(false)
  return (
    <>
    <h1>Real Example Ref</h1>
    <hr/>
    { show && <MultipleCustomHooks />}
    <button 
    className='btn btn-primary'
    onClick={ () => setShow(!show)}
    >ver/ocultar</button>
    </>
  )
}
