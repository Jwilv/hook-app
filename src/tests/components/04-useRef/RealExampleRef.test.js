import React from 'react';
import { shallow } from 'enzyme'
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';


describe('pruebas en el componente <RealExampleRef />', () => { 

    test('debe de mostrarse correctamente ', () => {
        const wrapper = shallow( <RealExampleRef />)
        expect(wrapper).toMatchSnapshot();
    });

    test('no deberia de mostrarse el componente <RealExampleRef /> ', () => {
        const wrapper = shallow( <RealExampleRef />)
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false )
    });

    test('debe de mostrarse el componente <RealExampleRef /> ', () => {
        const wrapper = shallow( <RealExampleRef />)
        wrapper.find('button').simulate('click')
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true )
    });
 })