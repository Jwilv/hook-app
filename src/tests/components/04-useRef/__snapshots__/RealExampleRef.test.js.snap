// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <RealExampleRef /> debe de mostrarse correctamente  1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    ver/ocultar
  </button>
</Fragment>
`;
